core/src/test/scala/integration/kafka/api/ProducerIdExpirationTest.scala

```diff
@@ -137,9 +137,6 @@ class ProducerIdExpirationTest extends KafkaServerTestHarness {
     producer.send(TestUtils.producerRecordWithExpectedTransactionStatus(topic1, 0, "4", "4", willBeCommitted = true))
     producer.send(TestUtils.producerRecordWithExpectedTransactionStatus(topic1, 0, "3", "3", willBeCommitted = true))
 
-    // Producer IDs should be retained.
-    assertEquals(1, producerState.size)
```

Comment:
One alternative I thought of is having this assertion just say the state is greater than 0. Did we see any flakes indicate the size was 0?

Reply:
Changed it to just an assertion to check greater than 0